crates/heat-sdk-cli/src/cli_commands/run/training.rs

```diff
@@ -0,0 +1,199 @@
+use clap::Parser;
+use colored::Colorize;
+use heat_sdk::{
+    client::{HeatClient, HeatClientConfig, HeatCredentials},
+    schemas::ProjectPath,
+};
+
+use crate::registry::Flag;
+use crate::{
+    commands::{execute_sequentially, BuildCommand, RunCommand, RunParams},
+    context::HeatCliContext,
+    generation::backend::BackendType,
+    logging::BURN_ORANGE,
+    print_info,
+};
+
+#[derive(Parser, Debug)]
+pub struct TrainingRunArgs {
+    /// The training functions to run
+    #[clap(short = 'f', long="functions", value_delimiter = ' ', num_args = 1.., required = true, help = "<required> The training functions to run. Annotate a training function with #[heat(training)] to register it."
+    )]
+    functions: Vec<String>,
+    /// Backend to use
+    #[clap(short = 'b', long = "backends", value_delimiter = ' ', num_args = 1.., required = true, help = "<required> Backends to use for training."
+    )]
+    backends: Vec<BackendType>,
+    /// Config files paths
+    #[clap(short = 'c', long = "configs", value_delimiter = ' ', num_args = 1.., required = true, help = "<required> Config files paths."
+    )]
+    configs: Vec<String>,
+    /// The Heat project ID
+    // todo: support project name and creating a project if it doesn't exist
+    #[clap(
+        short = 'p',
+        long = "project",
+        required = true,
+        help = "<required> The Heat project ID."
+    )]
+    project_path: String,
+    /// The Heat API key
+    #[clap(
+        short = 'k',
+        long = "key",
+        required = true,
+        help = "<required> The Heat API key."
+    )]
+    key: String,
+    /// Project version
+    #[clap(short = 't', long = "version", help = "The project version.")]
+    project_version: Option<String>,
+    /// The runner group name
+    #[clap(short = 'r', long = "runner", help = "The runner group name.")]
+    runner: Option<String>,
+}
+
+pub(crate) fn handle_command(args: TrainingRunArgs, context: HeatCliContext) -> anyhow::Result<()> {
+    match (&args.runner, &args.project_version) {
+        (Some(_), Some(_)) => remote_run(args, context),
+        (None, None) => local_run(args, context),
+        _ => Err(anyhow::anyhow!("Both runner and project version must be specified for remote run and none for local run.")),
```

Comment:
Not sure I like this implicit way of differentiating between local and remote using two different arguments.
Ideally local and remote should be decided by just the presence of a runner or not.

Maybe the pattern matching could be improved:
`Some(_), None` => "You must provide the project version to run on the runner with --project_version argument"
`None, Some(_)` => "Project version ignored when executing locally (i.e. no runner is defined with --runner argument"